### PR TITLE
Add debug info page (under /debug/rpcproxy/)

### PR DESCRIPTION
**Pull request by exekias**

### CODE REVIEW

**Comment by sixstone-qq** [comment]
I will have separate the part of Channel struct change in other commit. but not enough brave here

File: channels/channels.go [hunk 1]
```diff
@@ -117,6 +116,7 @@ func (VirtualConnections VirtualConnections) HandleOut(w http.ResponseWriter, r
 	log.Debug(pdu.Dump())
 	connection := VirtualConnections.GetOrCreate(pdu.Commands[1].(*packets.CookieCommand).Cookie)
 	connection.OutChannel.Cookie = pdu.Commands[2].(*packets.CookieCommand).Cookie
+	connection.Username, _, _ = r.BasicAuth() // BasicAuth requires Go >= 1.4
```
Comment (sixstone-qq) [inline channels/channels.go]: This deserves a commit :)

File: debug/debug.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	p[i], p[j] = p[j], p[i]
+}
+
+// EnableDebug info on proxy http server
+func EnableDebug(proxy *server.RPCProxy) {
+	prefix := "/debug/rpcproxy/"
+
+	http.HandleFunc(prefix, func(w http.ResponseWriter, r *http.Request) {
+		sections := []string{"info", "sessions"}
+		if err := indexTmpl.Execute(w, sections); err != nil {
```
Comment (sixstone-qq) [inline debug/debug.go]: Is it common in go to have two statements inside an if evaluation?

**Comment by sixstone-qq** [comment]
Apart from minor comments, :+1:

File: debug/debug.go (hunk 2 above)
Comment (exekias) [inline debug/debug.go]: It seems to be quite common, for error management only

**Pull request merged**